plugin/hosts: leaking (*time.Ticker) in goroutine inside periodicHostsUpdate

Reported by [Static Major from Orijtech Inc](https://github.com/marketplace/actions/staticmajor-analyzer), we can see this 
```shell
/coredns/plugin/hosts/setup.go:32:5: leaking resource created on line 28
```

which originates from https://github.com/coredns/coredns/blob/2f24b512d3999e9687ffbfaf8f8226c64185dca1/plugin/hosts/setup.go#L28 for which there is a missing `defer ticker.Stop()`

Kindly FYI-ing @willpoint @jhusdero @elias-orijtech @kirbyquerby

--- a/plugin/hosts/setup.go
+++ b/plugin/hosts/setup.go
@@ -26,6 +26,7 @@ func periodicHostsUpdate(h *Hosts) chan bool {
 
 	go func() {
 		ticker := time.NewTicker(h.options.reload)
+		defer ticker.Stop()
 		for {
 			select {
 			case <-parseChan:

